Reorganised structure for arduino 1.5+ and added library metadata

sgparry opened the pull request.

sgparry committed:
Reorganised folder structure for arduino 1.5+ and added library metadata files.

per1234 commented on .development:
```diff
@@ -0,0 +1 @@
+ 
```
What is the reason for adding this file?

sgparry replied:
It allows the library to be edited without complaint within the ordinary Arduino IDE - not 100% convninced
 myself whether or not it is a good idea to have it in as standard for source code repos or not - it either needs to be tracked or in .gitignore.

per1234 replied:
I often add a .development file to my own libraries for this reason, because I prefer to use the Arduino IDE for development work on the example sketches. However, I have it excluded from the repo.

There are two problems this file can cause:
- Libraries which contain this file are prohibited from being added to the Arduino Library Manager index: https://github.com/arduino/Arduino/wiki/Library-Manager-FAQ#how-is-the-library-list-generated
- It allows users to inadvertently modify the example sketch, causing confusion because they are accustomed to the library examples being immutable, known-good programs.

sgparry replied:
Foudn  guidance here: [https://arduino.github.io/arduino-cli/latest/library-specification/#development-flag-file](https://arduino.github.io/arduino-cli/latest/library-specification/#development-flag-file) that says not to push - so I will delete and add it to .gitignore.

sgparry replied:
I have made this change

per1234 commented on keywords.txt:
```diff
@@ -0,0 +1 @@
+freeMemory				KEYWORD2
```
This does not have the correct formatting. See the Arduino library specification for details:
https://arduino.github.io/arduino-cli/latest/library-specification/#keywordstxt-format

sgparry replied:
I was following a example from another library - have corrected

sgparry replied:
I have fixed this in line with the standard

per1234 commented on library.properties:
```diff
@@ -0,0 +1,9 @@
+name=MemoryFree
+version=1.0
```
Although this is valid, I recommend following semver:
https://semver.org/

sgparry replied:
So int this case would you recommend 1.0.0 ? I think the behaviour is stable enough to define a public release.

per1234 replied:
Sounds good to me.

sgparry replied:
I have made this change

per1234 commented on library.properties:
```diff
@@ -0,0 +1,9 @@
+author=Various; see https://forum.arduino.cc/index.php?topic=27536.0
+maintainer=Neil McNeight (http://mcneight.org/)
+sentence=Functions for calculating amount for free data memory (RAM).
+paragraph=Functions for calculating amount for free data memory (RAM).
```
The `paragraph` value is typically displayed immediately after the `sentence` value, so duplicating the `sentence` content in `paragraph` has a harmful effect. It should either contain additional unique text or else be left blank.

sgparry replied:
I will amend to add further detail.

sgparry replied:
I have made this change

sgparry committed:
Corrections in line with latest arduino library standards

sgparry committed:
Attempt fix stalling travis build

sgparry committed:
travis: TinyWireM already included in install.

per1234 committed:
Correct board ID for Zero in Travis CI configuration

Incorrect FQBN causes the Arduino IDE to attempt to start the GUI, which results in a hang and eventual timeout on the headless Travis CI machine.

sgparry committed:
Merge pull request #1 from per1234/Library1.5

Correct board ID for Zero in Travis CI configuration
